src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -259,18 +259,22 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     //   connection, then this'll revoke the existing connection's handlers.
     void ControlCore::Connection(const TerminalConnection::ITerminalConnection& newConnection)
     {
-        if (_connection)
+        const bool replacing = _connection != nullptr;
+        if (replacing)
         {
             _connectionOutputEventRevoker.revoke();
-
-            // Fire off a connection state changed notification, to let our hosting
-            // app know that we're in a different state now.
-            if (newConnection.State() != _connection.State())
+        }
+        // Bail early if we're not actually giving the Terminal a new connection
+        if (newConnection == nullptr)
+        {
+            if (replacing)
             {
+                _connection.Close();
                 _ConnectionStateChangedHandlers(*this, nullptr);

```

Comment:
this is not necesasry if the connection is properly set up. We may want to just unsubscribe and destruct...

Also, I think you'll leave us subscribed to the event on the old outgoing connection on line R279